### PR TITLE
[WFCORE-1985] - :read-resource-description operation should consisten…

### PR DESCRIPTION
**Pull request by JiriOndrusek**
…tly order attributes by alphabet

Issue: https://issues.jboss.org/browse/WFCORE-1985

### CODE REVIEW

**Comment by jmesnil** [comment]
retest this please

**Comment by jmesnil** [comment]
retest this please

**Comment by bstansberry** [comment]
This is about attribute ordering (which is fine) but the JIRA is about 'resource' order, which I figured meant children.

**Comment by JiriOndrusek** [comment]
@bstansberry You are correct. I'll try to add also a  fix for order of children.

File: controller/src/main/java/org/jboss/as/controller/descriptions/DefaultResourceDescriptionProvider.java [hunk 1]
```diff
@@ -119,7 +120,17 @@ public ModelNode getModelDescription(Locale locale) {
         // Sort the attribute descriptions based on attribute group and then attribute name
         Set<String> attributeNames = registration.getAttributeNames(PathAddress.EMPTY_ADDRESS);
 
-        Map<AttributeDefinition.NameAndGroup, ModelNode> sortedDescriptions = new TreeMap<>();
+        //fix of WFCORE-1985 - attributes have to be sorted firstly by name
+        //case that two attributes of the same name are allowed is not possible, see error WFLYCTL0043
+        Comparator<AttributeDefinition.NameAndGroup> comparator = (x, y) -> {
```
Comment (jmesnil) [inline controller/src/main/java/org/jboss/as/controller/descriptions/DefaultResourceDescriptionProvider.java]: Could you please remove this lambda et replace it by an anonymous class please?

**Comment by jmesnil** [comment]
The author of the JIRA issue mentions the `/subsystem=iiop-openjdk` resource and this resource has no children but many attributes (as shown by [wildscribe](https://wildscribe.github.io/WildFly/11.0/subsystem/iiop-openjdk/index.html).

I think your PR is fine (notwistanding my change request) but the JIRA description and title should be updated to mention that this is about reordering of *attributes*.
As @bstansberry mentioned, ordering of resources is otherwise understood as orderding of *children*  resources.

File: controller/src/main/java/org/jboss/as/controller/descriptions/DefaultResourceDescriptionProvider.java (hunk 1 above)
Comment (JiriOndrusek) [reply controller/src/main/java/org/jboss/as/controller/descriptions/DefaultResourceDescriptionProvider.java]: @jmesnil it's done, I also renamed PR and issue to contain attributes as requested below

**Comment by jmesnil** [comment]
retest this please

**Pull request merged**